envoy/router/shadow_writer.h

```diff
@@ -27,6 +27,18 @@ class ShadowWriter {
    */
   virtual void shadow(const std::string& cluster, Http::RequestMessagePtr&& request,
                       const Http::AsyncClient::RequestOptions& options) PURE;
+
+  /**
+   * Initialize shadowing a request. Differs from the above in that additional
+   * data can be passed to the returned handle after the headers have been sent.
+   * @param cluster supplies the cluster name to shadow to.
+   * @param headers supplies the headers for initializing the shadow.
+   * @param timeout supplies the shadowed request timeout.
```

Comment:
It's not just a timeout, it's request options instead. Can you update this and the declaration above?